spring-cloud-skipper-shell-commands/src/main/java/org/springframework/cloud/skipper/shell/config/HelpAwareShellApplicationRunner.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.cloud.skipper.shell.config;
+
+import java.io.BufferedReader;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.Reader;
+
+import org.springframework.boot.ApplicationArguments;
+import org.springframework.boot.ApplicationRunner;
+import org.springframework.core.annotation.Order;
+import org.springframework.shell.jline.DefaultShellApplicationRunner;
+import org.springframework.util.Assert;
+import org.springframework.util.FileCopyUtils;
+
+/**
+ * {@link ApplicationRunner} that print a help-text to the console in case
+ * the help command-line option was provided and will also exit the application in that case.
+ *
+ * @author Gunnar Hillert
+ *
+ * @see DefaultShellApplicationRunner
+ * @see InitialConnectionApplicationRunner
+ */
+@Order(DefaultShellApplicationRunner.PRECEDENCE - 20)
+public class HelpAwareShellApplicationRunner implements ApplicationRunner {
```

Comment:
This could possibly be on a separate PR as this is one other nice feature and can be tracked easily as a separate commit. Will try to separate out when merging, if not will be under the same one.